modules/in_band_dtmf/in_band_dtmf.c

```diff
@@ -0,0 +1,324 @@
+/**
+ * @file in_band_dtmf.c  DTMF decoder
+ */
+#include <time.h>
+#include <ctype.h>
+#include <string.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+#include "audio.h"
+
+
+/**
+ * @defgroup in_band_dtmf in_band_dtmf
+ *
+ * Audio filter that de- and encodes DTMF tones.
+ *
+ * The tone length of an encoded tone is 0.1s.
+ * New tones can be added while encoding is active.
+ */
+
+struct in_band_dtmf_filt_dec {
+	struct aufilt_dec_st af;  /* inheritance */
+	struct dtmf_dec *dec;
+	const struct audio *au;
+	struct le le_priv;
+	struct tmr tmr_dtmf_end;
+};
+
+
+struct in_band_dtmf_filt_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct mbuf *mb;
+	unsigned srate;
+	struct le le_priv;
+};
+
+
+static struct list encs;
+static struct list decs;
+
+
+static void dtmfend_handler(void *arg)
+{
+	char digit = (char)(uintptr_t)arg;
+	struct in_band_dtmf_filt_dec *st;
+
+	if (list_isempty(&decs)) {
+		warning("in_band_dtmf: no active call\n");
+		return;
+	}
+
+	st = decs.head->data;
+
+	st->au->eventh(digit, true, st->au->arg);
```

Comment:
Perhaps it is better to add a wrapper function for this in audio.c ?

Reply:
I introduced now a `audio_get_event_handler` and also a `audio_get_argument`. 

This could be replaced by a `audio_call_event_handler(struct audio *au, int key, bool end)`.
In this case, the `au->arg` argument would be used for `au->eventh` arg argument.
But i'm not aware about the meaning of this argument. Is this every time the `au->arg` or can it be different?